### PR TITLE
Added Azure.AppConfig.PurgeProtect

### LINKED ISSUE #1689: Enable purge protection for App Configuration stores
# Rule request

## Suggested rule change

App Configuration supports purge protection to extend the protection provided by soft-delete. Purge protection limits data loss causes by accidental and malicious purges of deleted configuration stores by enforcing an mandatory retention interval.

This feature only applies to Standard SKU configuration stores. Free configuration stores should be ignored by this rule.

This is enabled by setting the `properties.enablePurgeProtection` property to `true`.

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.AppConfiguration/configurationStores**

## Additional context

[Azure deployment reference](https://learn.microsoft.com/azure/templates/microsoft.appconfiguration/configurationstores)
[Purge protection](https://learn.microsoft.com/azure/azure-app-configuration/concept-soft-delete#purge-protection)

Related rules include:

- https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.KeyVault.PurgeProtect/

### PR DESCRIPTION
**Pull request by bengeset96**
## PR Summary

Fixes #1689 

Added Azure.AppConfig.PurgeProtect rule for ensuring that app configuration stores with supporting SKU has it enabled.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by bengeset96** [comment]
@BernieWhite 

Ready for review.

File: tests/PSRule.Rules.Azure.Tests/Resources.AppConfig.json [hunk 1]
```diff
@@ -396,14 +397,15 @@
       "encryption": {
         "keyVaultProperties": null
       },
-      "privateEndpointConnections": null
+      "privateEndpointConnections": null,
+      "enablePurgeProtection": true
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Resources.AppConfig.json:401]: Let's have one case with `enablePurgeProtection` set to `false`.

**Review by BernieWhite — CHANGES_REQUESTED** [review]
@bengeset96 Great work. Only one minor suggestion to enhance the test cases.

File: tests/PSRule.Rules.Azure.Tests/Resources.AppConfig.json (hunk 1 above)
Comment (bengeset96) [reply tests/PSRule.Rules.Azure.Tests/Resources.AppConfig.json]: Agreed.

**Comment by bengeset96** [comment]
@BernieWhite 

Added one case with `enablePurgeProtection` set to `false`.

**Review by BernieWhite — APPROVED** [review]
Awesome! Thanks @bengeset96 all good to merge.

**Pull request merged**